src/query-builder/DeleteQueryBuilder.ts

```diff
@@ -231,9 +232,13 @@ export class DeleteQueryBuilder<Entity> extends QueryBuilder<Entity> implements
         const whereExpression = this.createWhereExpression();
         const returningExpression = this.createReturningExpression();
 
-        if (returningExpression && (this.connection.driver instanceof PostgresDriver || this.connection.driver instanceof CockroachDriver)) {
+        if (
+            returningExpression &&
+            (this.connection.driver instanceof PostgresDriver ||
+                this.connection.driver instanceof CockroachDriver ||
+                this.connection.driver instanceof MysqlDriver)
```

Comment:
don't we need to check for `connection.driver.options.type === "mariadb"` ? (instead of this.connection.driver instanceof MysqlDriver)

Reply:
It doesn't work with MySQL, only MariaDB. It's ok?

Reply:
yes, that's why `connection.driver.options.type === "mariadb"` check will work better then checking for instanceof here

Reply:
> don't we need to check for `connection.driver.options.type === "mariadb"` ? (instead of this.connection.driver instanceof MysqlDriver)

Opps... I didn't see the question mark above.
I check MariaDB [this line](https://github.com/typeorm/typeorm/pull/8673/files#diff-a5db73ecf35a00a64edc5bc64385dd06bcf618736cc3d9fd422091a178ffcf82R878-R879). If try to use `returning()`, will get an error if it is not MariaDB here.
So `instanceof MysqlDriver` may be left for later when MySQL supports `returning()`.